### PR TITLE
Add uvloop to python rosdep index

### PR DESCRIPTION
**Pull request by nachovizzo**
Please add the following dependency to the rosdep database.

## Package name:

python3-uvloop

## Package Upstream Source:

https://github.com/MagicStack/uvloop

## Purpose of using this:

Dependency to be used by the new version of `ament-black`. `uvtool` is required to fix a concurrency problem on `--reformat`, this was solved in https://github.com/botsandus/ament_black/pull/4

Distro packaging links:

## Links to Distribution Packages

- Debian: https://packages.debian.org/
  -[ REQUIRED](https://packages.debian.org/bookworm/python3-uvloop)
- Ubuntu: https://packages.ubuntu.com/
  - https://packages.ubuntu.com/jammy/python3-uvloop
- Fedora: https://packages.fedoraproject.org/
  - https://packages.fedoraproject.org/pkgs/python-uvloop/python3-uvloop/
- Arch: https://www.archlinux.org/packages/
  - https://archlinux.org/packages/extra/x86_64/python-uvloop/
- Gentoo: https://packages.gentoo.org/
  - https://packages.gentoo.org/packages/dev-python/uvloop
- macOS: https://formulae.brew.sh/
  - NOT AVAILABLE
- Alpine: https://pkgs.alpinelinux.org/packages
  - NOT AVAILABLE
- NixOS/nixpkgs: https://search.nixos.org/packages
  - https://search.nixos.org/packages?channel=23.05&from=0&size=50&sort=relevance&type=packages&query=uvloop
- openSUSE: https://software.opensuse.org/package/
  - https://software.opensuse.org/package/python-uvloop
- rhel: https://rhel.pkgs.org/
  - Ihttps://pkgs.org/search/?q=uvloop

### CODE REVIEW

**Comment by nachovizzo** [comment]
So, [this error](https://github.com/ros/rosdistro/actions/runs/6489344649/job/17623432790?pr=38754#step:6:404) says:

image


BUT! As stated in the PR description, I can see from [the arch repos](https://archlinux.org/packages/extra/x86_64/python-uvloop/)

image


**How can I fix this**?

**Comment by nachovizzo** [comment]
The same is happening for ubuntu... and all the other distros 
image

**Comment by nachovizzo** [comment]
@quarkytale  @tfoote 

Could we merge this so I can continue with the release of #38753. Once uvloop is in rosdep I will also release the new version of `ament-black` for rolling and humble as well

**Review by tfoote — APPROVED** [review]
This looks good to me. It will need a second review likely from our reviewer rotation.

**Review by quarkytale — APPROVED** [review]

**Pull request merged**
